### PR TITLE
Implement Password reset

### PR DESCRIPTION
**Pull request by Endowmissy**
#### What does this PR do?
allows user to reset their password successfully

#### How should this manually be Tested?
- `git clone` __https://github.com/devcareer/Dae-Alright-Backend__
- `cd` **Dae-Alright-Backend**
- `git checkout <branch-name>`
- `npm install`
- `npm test`

### CODE REVIEW

**Comment by maestro-1** [comment]
Resolve the conflict first, please. The conflict from auth.controller.js

File: package.json [hunk 1]
```diff
@@ -36,7 +36,9 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.6.3",
+    "@sendgrid/mail": "^6.4.0",
     "bcrypt": "^3.0.6",
+    "bcryptjs": "^2.4.3",
```
Comment (TundeMercy) [inline package.json:41]: Why bcryptjs again?

File: src/controllers/password.controller.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+ * @param {obj} req The request object
+ * @param {obj} res The response object
+ * @returns {json} The response from db or error.
+ */
+export const forgotPassword = async (req, res) => {
+  const { email } = req.body;
+
+  const user = await User.findOne({ where: { email } });
+  if (!user) {
+    return errorResponse(res, 400, 'Email address does not match any account');
```
Comment (TundeMercy) [inline src/controllers/password.controller.js]: This should be a `404` error I suppose

File: src/controllers/password.controller.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    text: `Hi ${user.firstName} \n 
+    A request to reset your account password was receieved. Please click on the following link ${url} to reset your password. \n\n 
+    If you did not request this, please ignore this email and your password will remain unchanged.\n`,
+  };
+
+  try {
+    sgMail.send(mailOptions);
+    return successResponse(res, 200, 'A reset link has been sent to your email');
+  } catch (error) {
+    return errorResponse(res, 400, 'An error occured');
```
Comment (TundeMercy) [inline src/controllers/password.controller.js]: Please change the error code from 400, this is obviously not an error caused by a bad request from user

File: src/controllers/password.controller.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  const { token } = req.params;
+  const hashedPassword = await bcrypt.hash(password, 10);
+  const user = await jwt.verify(token, process.env.JWT_KEY, (err, decoded) => {
+    if (err) {
+      return errorResponse(res, 400, 'Invalid token, please provide a valid token');
+    }
+    return decoded;
+  });
+  if (!user) {
+    return errorResponse(res, 400, 'Password reset token is invalid or might has expired');
```
Comment (TundeMercy) [inline src/controllers/password.controller.js]: `might **have** expired` you mean

**Review by TundeMercy — COMMENTED** [review]
@Endowmissy, please check all comments

File: package.json (hunk 1 above)
Comment (Endowmissy) [reply package.json:41]: The bcrypt was not working for me, reason for bcryptjs

File: src/controllers/password.controller.js (hunk 2 above)
Comment (Endowmissy) [reply src/controllers/password.controller.js]: okay

File: src/controllers/password.controller.js (hunk 3 above)
Comment (Endowmissy) [reply src/controllers/password.controller.js]: okay

File: src/controllers/password.controller.js (hunk 4 above)
Comment (Endowmissy) [reply src/controllers/password.controller.js]: typo error, will fix that

**Review by meetKazuki — APPROVED** [review]
Here's to hoping it doesn't bite us later🍺

**Comment by meetKazuki** [comment]
[Password Reset Feature](https://trello.com/c/RoQ7DnMR/9-password-reset-feature)

**Pull request merged**